manifests/compose.pp

```diff
@@ -21,17 +21,11 @@
 #   Proxy to use for downloading Docker Compose.
 #
 class docker::compose(
-  $ensure = 'present',
-  $version = $docker::params::compose_version,
-  $install_path = $docker::params::compose_install_path,
-  $proxy = undef
+  Optional[Pattern[/^present$|^absent$/]] $ensure                                                                         = 'present',
+  Optional[String] $version                                                                                               = $docker::params::compose_version,
+  Optional[String] $install_path                                                                                          = $docker::params::compose_install_path,
+  Optional[Pattern[/^(https?:\/\/)?([^:^@]+:[^:^@]+@|)([\da-z\.-]+)\.([\da-z\.]{2,6})(:[\d])?([\/\w \.-]*)*\/?$/]] $proxy = undef
```

Comment:
I have to scroll to see what the parameter names are, maybe a custom type alias is a good idea here

Reply:
Same comment for every large data type

Reply:
@jeefberkey Thank you for the valuable comment

Reply:
@jeefberkey The large data types are not repeating alot to create a data type.So will specify the assert_type function for all large data types outside the variable declaration area so keep it short.